ARCs/arc-draft_alternative_0014.md

```diff
@@ -0,0 +1,283 @@
+
+## Motivation
+
+Traditional systems enforce SSO login leveraging on authentication of users with *credentials*, i.e. username and password. Once the identity of a user is verified, the system establishes an authenticated session. Sessions allows users to interact with the system without authenticate themselves anytime. Sessions are represented with the so called `session-id` (typically a cookie or a JSON Web Token - JWT).
+
+In a blockchain context, users do not have credentials. Conversely, in the Web3 users are identified through their unique blockchain public addresses (accounts for Algorand). Therefore, traditional authentication mechanisms result impractical. To interact with a dApp, users connect their wallet to the application. Using an authentication layer based on credentials on top of that mechanism might result redundant and inefficient.
+
+In Web3, dApps and traditional systems will be increasingly more interconnected. It is not difficult to imagine users consuming services both from a dApp and a traditional system simultaneously. In that case, a user should be authenticated first through traditional SSOs, and then connecting their blockchain wallet. A better approach should enforce a single SSO procedure.
+
+This ARC provides the standard to authenticate users leveraging on their Algorand accounts, and without relying on credentials verification.
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+> Comments like this are non-normative.
+
+Interfaces are defined in TypeScript. All the objects that are defined are valid JSON objects, and all JSON string types are UTF-8 encoded.
+
+This ARC uses interchangeably the terms "*blockchain address*", "*public key*", and "*PK*" to indicate the on-chain address of a blockchain identity, and in particular of an Algorand account.
+
+### Overview
+
+This document describes a standard approach to authenticate Users with their blockchain identity. In particular, in this ARC Users' Algorand address are used as a *unique identifiers*, and the secret key of that address is used to digitally sign a *message* as a proof of identity for a  Verifier. Hence, upon signature verification, the Verifier initializes an authenticated session with the User by issuing a new `session-id`.
+
+To sum up, given an Algorand account <*PKa, SKa*>, this ARC defines the standards for:
+
+- create an ARC-0014 compliant digital signature with *SKa*;
+- verify an ARC--014 compliant digital signature with *PKa*.
+
+### Assumptions
+
+The standard proposed in this document works under the following assumptions:
+
+- User and Verifier communicates over secure SSL/TLS encrypted channels;
```

Comment:
TLS 1.3 supports ed25519 and raw keys for both client and server . It might be worth looking into just using TLS and client authentication on the level of the transport. If wallets can offer the key via eg something like PKCS#11 then this might be an alternative worth considering.